persist/fs/commitlog/types.go

```diff
@@ -162,4 +163,11 @@ type Options interface {
 
 	// BytesPool returns the checked bytes pool
 	BytesPool() pool.CheckedBytesPool
+
+	// SetShardTimeRanges sets the ShardTimeRanges that determines which
+	// commitlogs to include in the bootstrap
+	SetShardTimeRanges(value result.ShardTimeRanges) Options
+
+	// ShardTimeRanges returns the ShardTimeRanges
+	ShardTimeRanges() result.ShardTimeRanges
```

Comment:
If these are only used during Bootstrap, could you move them to`storage/bootstrap/bootstrapper/commitlog`

Reply:
I tried doing that and then I realized that persis/fs/commit_log.go has an Iter() method which would mean that I would also need to pass the bootstrapping options into the CommitLog itself which seems weird

Reply:
Shouldn't introduce a dependency on the `result` package here. Here's an alternative: you can get this information using a new ctor for `Iterator` which takes a list of times (min/max if that's easier), or a predicate which returns a bool indicating if a commit log block start time should be read. 

What do you think?

Reply:
New ctor for Iterator seems reasonable, although it leaves open the question of how to handle the Iter method in commit_log.go:

```
func (l *commitLog) Iter() (Iterator, error) {
	return NewIterator(l.opts)
}
```

Do I just make that method accept the min/max as well? I actually wouldn't be opposed t removing that method since it barely does anything